### PR TITLE
Add support for forcing tool use when you programatically

### PR DESCRIPTION
**Pull request by IanCal**

### CODE REVIEW

**Comment by IanCal** [comment]
Note - I tried adding tests about the error, but had the same issue as before with uncaught exceptions because we throw in two ways. I'll put up a PR with a clearer case where this happens but didn't want to hold this up because of that.

File: packages/ai-bot/helpers.ts
Comment (lukemelia) [inline packages/ai-bot/helpers.ts:344]: Is this recoverable, or does it "break" a room forever?

**Review by lukemelia — APPROVED** [review]

File: packages/ai-bot/helpers.ts
Comment (IanCal) [reply packages/ai-bot/helpers.ts:344]: I'll test this before merging, with the check on the host we shouldn't get into this situation anyway but with the current setup this should be temporary as the list of commands and the requiredtoolcall is just on the last user message so the next valid one sent will be fine

**Pull request merged**
